mu/app.py

```diff
@@ -88,22 +136,62 @@ class StartupWorker(QObject):
     The animated splash screen will be shown until this thread is finished.
     """
 
-    finished = pyqtSignal()
+    finished = pyqtSignal()  # emitted when successfully finished.
+    failed = pyqtSignal(str)  # emitted if finished with an error.
+    display_text = pyqtSignal(str)  # emitted to update the splash text.
 
     def run(self):
         """
         Blocking and long running tasks for application startup should be
         called from here.
         """
-        venv.ensure_and_create()
-        self.finished.emit()  # Always called last.
+        try:
+            venv.ensure_and_create(self.display_text)
+            self.finished.emit()  # Always called last.
+        except Exception as ex:
+            # Catch all exceptions just in case.
+            # Report the failure, along with a summary to show the user.
+            stack = traceback.extract_stack()[:-1]
+            msg = "\n".join(traceback.format_list(stack))
+            msg += "\n\n" + traceback.format_exc()
+            self.failed.emit(msg)
+            # Sleep a while in the thread so the user sees something is wrong.
+            time.sleep(7)
+            self.finished.emit()
+            # Re-raise for crash handler to kick in.
+            raise ex
+        finally:
+            # Always clean up the startup splash logging handlers.
+            while vlogger.hasHandlers() and vlogger.handlers:
+                handler = vlogger.handlers[0]
+                if isinstance(handler, SplashLogHandler):
+                    vlogger.removeHandler(handler)
```

Comment:
If one of the handlers is not `SplashLogHandler`, and so it's not removed, would this turn into an infinite loop?

Reply:
Yes and no. But this code, as I've written it could be quite brittle so you're right to point this out (thank you..!).

I'll change it so ALL log handlers are removed, since the `vlogger` is only ever used during splash start-up.